pkg/kindsys/report.go

```diff
@@ -53,47 +55,123 @@ var plannedCoreKinds = []string{
 	"QueryHistory",
 }
 
+type Kind struct {
+	Name     string `json:"name"`
+	Type     string `json:"type"`
+	Maturity string `json:"maturity"`
+}
+
 type KindStateReport struct {
-	Core       []kindsys.CoreStructuredProperties `json:"core"`
-	Raw        []kindsys.RawProperties            `json:"raw"`
-	Composable []kindsys.ComposableProperties     `json:"composable"`
+	Kinds      map[string]Kind      `json:"kinds"`
+	Dimensions map[string]Dimension `json:"dimensions"`
+}
+
+type Dimension map[string]DimensionValue
+
+type DimensionValue struct {
+	Name  string   `json:"name"`
+	Items []string `json:"items"`
+	Count int      `json:"count"`
 }
 
+// emptyKindStateReport is used to ensure certain
+// dimension values are present (even if empty) in
+// the final report.
 func emptyKindStateReport() KindStateReport {
 	return KindStateReport{
-		Core:       make([]kindsys.CoreStructuredProperties, 0),
-		Raw:        make([]kindsys.RawProperties, 0),
-		Composable: make([]kindsys.ComposableProperties, 0),
+		Kinds: make(map[string]Kind),
+		Dimensions: map[string]Dimension{
+			"maturity": {
+				"merged":       emptyDimensionValue("merged"),
+				"experimental": emptyDimensionValue("experimental"),
+				"stable":       emptyDimensionValue("stable"),
+				"mature":       emptyDimensionValue("mature"),
+			},
+			"type": {
```

Comment:
per [glossary](https://docs.google.com/document/d/13Rv395_T8WTLBgdL-2rbXKu0fx_TW-Q9yz9x6oBjm6g/edit#heading=h.ff56w75l80de)
```suggestion
			"category": {
```